src/Components/Address/NewAddressForm.tsx

```diff
@@ -0,0 +1,261 @@
+import { Button, Column, GridColumns, Input, Message } from "@artsy/palette"
+import { Formik, Form, FormikConfig, useFormikContext } from "formik"
+import { FC, useEffect } from "react"
+import { CountrySelect } from "Components/CountrySelect"
+
+export interface AddressFormValues {
+  name?: string
+  country?: string
+  addressLine1?: string
+  city?: string
+  region?: string
+  postalCode?: string
+  addressLine2?: string
+}
+
+export type AddressChangeHandler = (
+  address: AddressFormValues,
+  key: keyof AddressFormValues
+) => void
+
+export const EMPTY_ADDRESS = {
+  name: "",
+  country: "",
+  addressLine1: "",
+  addressLine2: "",
+  city: "",
+  postalCode: "",
+  region: "",
+}
+
+export const DEFAULT_VALIDATION_SCHEMA = Yup.object().shape({
+  name: Yup.string().required("Name is required"),
+  country: Yup.string().required("Country is required"),
+  addressLine1: Yup.string().required("Address is required"),
+  city: Yup.string().required("City is required"),
+  region: Yup.string().required("Region is required"),
+  postalCode: Yup.string().required("Postal Code is required"),
+})
+
+// For debugging
+export const BasicAddressForm: FC<{
+  onChange?: (args: any) => void
+}> = props => {
+  const { values, errors, touched } = useFormikContext<AddressFormValues>()
+
+  const serializedState = JSON.stringify({ values, errors, touched })
+  // hack to inspect/debug intermediate form state before submission
+  useEffect(() => {
+    props.onChange && props.onChange({ values, errors, touched })
+
+    // really props.onChange should be wrapped in useCallback before being passed in - then we would have this guarantee
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [serializedState])
+  return (
+    <Form>
+      <AddressInputs
+        endContent={
+          <Button type="submit" variant="primaryBlack" width="50%">
+            Save and Continue
+          </Button>
+        }
+      />
+    </Form>
+  )
+}
+
+export interface AddressFormWrapperProps<T extends AddressFormValues>
+  extends Omit<FormikConfig<T>, "initialValues"> {
+  initialValues: Partial<T>
+}
+
+/**
+ * A Formik context that provides a default validation schema and initial values
+ * for an address form. Provide a custom type extending `AddressFormValues` to
+ * add additional fields, along with other Formik props. The children should
+ * formik-compatible contain inputs for any AddressFormValues and other fields.
+ */
+export const AddressFormWrapper = <
+  T extends AddressFormValues = AddressFormValues
+>({
+  initialValues = {} as Partial<T>,
+  children,
+  validate,
+  validationSchema,
+  ...props
+}: AddressFormWrapperProps<T>): JSX.Element => {
+  let validationRule: Pick<FormikConfig<T>, "validate" | "validationSchema">
+  if (validationSchema) {
+    validationRule = { validationSchema }
+  } else if (validate) {
+    validationRule = { validate }
+  } else {
+    validationRule = { validationSchema: DEFAULT_VALIDATION_SCHEMA }
+  }
```

Comment:
Whats the difference between `validationSchema` and `validate` and the default case? 

Couldn't we just do something with standard default props, like 
```tsx
const AddressForm = ({ validationSchema = DEFAULT_VALIDATION_SCHEMA, ... }) => { ... } 
```